core/templates/domain/statistics/playthrough-backend-api.service.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend API service for playthrough.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Playthrough } from
+  'domain/statistics/PlaythroughObjectFactory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+interface IStorePlaythroughBackendResponse {
+  'playthrough_stored': boolean;
+  'playthrough_id': string;
+}
+
+export class StorePlaythroughResponse {
+  playthroughStored: boolean;
+  playthroughId: string;
+
+  constructor(playthroughStored: boolean, playthroughId: string) {
+    this.playthroughId = playthroughId;
+    this.playthroughStored = playthroughStored;
+  }
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughBackendApiService {
+  STORE_PLAYTHROUGH_URL: string = (
+    '/explorehandler/store_playthrough/<exploration_id>');
+
+  constructor(
+    private http: HttpClient,
```

Comment:
Please indent these by 4 spaces (see difference between this and line 54)

Reply:
done